### PR TITLE
Keadm: optimize OS detect

### PR DESCRIPTION
**Pull request by daixiang0**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Now we use OS release to detect OS then we need to add specific OS name by hand, while we only want to know which package manager the OS uses, so optimize it.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: keadm/cmd/keadm/app/cmd/util/common.go [hunk 1]
```diff
@@ -116,28 +114,33 @@ func (co *Common) SetOSInterface(intf types.OSTypeInstaller) {
...
-		return &CentOS{}
+	switch GetPackageManager() {
+	case APT:
+		return &DebOS{}
+	case YUM:
+		return &RpmOS{}
 	default:
-		fmt.Printf("This OS version is currently un-supported by keadm, %s", GetOSVersion())
-		panic("This OS version is currently un-supported by keadm,")
+		fmt.Println("Failed to detect apt/yum command, exit")
```
Comment (fisherxu) [inline keadm/cmd/keadm/app/cmd/util/common.go:142]:
```suggestion
		fmt.Println("Failed to detect package manager command, exit")
```
for supporting more types?

Comment (daixiang0) [reply keadm/cmd/keadm/app/cmd/util/common.go]: Maybe in other prs, `apt` and `yum` should cover most use cases.

File: keadm/cmd/keadm/app/cmd/util/common.go [hunk 2]
```diff
@@ -42,21 +42,28 @@ import (
...
+	KubeEdgeConfigModulesYaml    = KubeEdgeConfPath + "/modules.yaml"
+
+	KubeEdgeCloudCertGenPath     = KubeEdgePath + "certgen.sh"
+	KubeEdgeEdgeCertsTarFileName = "certs.tgz"
+	KubeEdgeCloudConfPath        = KubeEdgePath + "kubeedge/cloud/conf"
+	KubeEdgeCloudCoreYaml        = KubeEdgeCloudConfPath + "/controller.yaml"
+	KubeEdgeCloudCoreModulesYaml = KubeEdgeCloudConfPath + "/modules.yaml"
+	KubeCloudBinaryName          = "cloudcore"
+
+	KubeEdgeConfigDir     = KubeEdgePath + "config/"
```
Comment (fisherxu) [inline keadm/cmd/keadm/app/cmd/util/common.go:66]: Delete the unused consts

File: keadm/cmd/keadm/app/cmd/util/common.go [hunk 3]
```diff
@@ -96,28 +95,33 @@ func (co *Common) SetOSInterface(intf types.OSTypeInstaller) {
 	co.OSTypeInstaller = intf
 }
 
-//GetOSVersion gets the OS name
-func GetOSVersion() string {
-	cmd := NewCommand("source /etc/os-release && echo $ID")
+//GetPackageManager get package manager of OS
+func GetPackageManager() string {
+	cmd := NewCommand("command -v apt || command -v yum")
```
Comment (gccio) [inline keadm/cmd/keadm/app/cmd/util/common.go:100]: if `apt` is a alias, for example `alias apt="apt -y"`, `command -v apt` may not end with `apt`. 

```shell
$ docker run --rm -it ubuntu
root@91d3467eecaf:/# command -v apt
/usr/bin/apt
root@91d3467eecaf:/# alias apt="apt -y"
root@91d3467eecaf:/# command -v apt
alias apt='apt -y'
root@91d3467eecaf:/# which apt
/usr/bin/apt
root@91d3467eecaf:/#
```
maybe we could try with `which`

Comment (gccio) [reply keadm/cmd/keadm/app/cmd/util/common.go:100]: We can find the package manager under `/usr/bin/`, most package manager is under this path. And add a parameter in keadm, like `--package-manager` or other. User can provide their package manager to keadm manual.

If keadm have this option, we can provide a interface to user, for example `--package-manage=MYSCRIPT`. The script implement some func like `install mqtt`, . we just use `MYSCRIPT installl mqtt` to install mqtt and other things. user need to complete the install logic, install online or offline, all is ok. :)

Comment (daixiang0) [reply keadm/cmd/keadm/app/cmd/util/common.go:100]: Refer to https://stackoverflow.com/questions/592620/how-can-i-check-if-a-program-exists-from-a-bash-script, `which` is not a good choice.

An alias does not make whole install cmd like `apt install -y mqtt` failed, besides, we use `sudo apt`, in most production environments, there is little alias in root account. If exists, fail output would raise error to notice users.

We can auto detect package manager cmd, no need to ask users to specify it. Also, install by  package manager still need network in most cases, it is outside this pr indeed.

**Review by gccio — COMMENTED** [review]
/lgtm

**Pull request merged**